[faucet]Restrict writes to a particular user

ashishb opened the pull request:
### Description

Restrict Firebase writes to a particular user and in the web login using that
user in Firebase before writing to the database.

### Tested

TBD

### Related issues

- Fixes #229

ashishb commented:
> Are these getting stored in the encrypted config file?
Yes

> Also feels a little brittle in terms of relying on one username/pass combo.
That's what we do for the admin account as well. Any better recommendations?

trianglesphere commented:
> > > Also feels a little brittle in terms of relying on one username/pass combo.

> > That's what we do for the admin account as well. Any better recommendations?

Unfortunately not really.

ashishb committed:
[faucet]Restrict writes to a particular user

Restrict Firebase writes to a particular user and in the web login using that
user in Firebase before writing to the database.

Testing

1.  `yarn firebase login`
2.  `yarn deploy:staging`
3.  Deployment can be seen at [https://console.firebase.google.com/project/celo-faucet-staging/overview](https://console.firebase.google.com/project/celo-faucet-staging/overview)
4.  You can simulate the access at [https://console.firebase.google.com/project/celo-faucet-staging/database/celo-faucet-staging/rules](https://console.firebase.google.com/project/celo-faucet-staging/database/celo-faucet-staging/rules)

`packages/web $ yarn run dev`
Go to [http://localhost:3000/build/wallet](http://localhost:3000/build/wallet) and perform submit, verify that no failure appears in the logs.
